src/vs/workbench/contrib/terminal/browser/terminalActions.ts

```diff
@@ -128,244 +129,225 @@ export class TerminalLaunchHelpAction extends Action {
+		title: { value: localize('workbench.action.terminal.focusNextPane', "Focus Next Terminal in Terminal Group"), original: 'Focus Next Terminal in Terminal Group' },
+		keybinding: {
+			primary: KeyMod.Alt | KeyCode.RightArrow,
+			secondary: [KeyMod.Alt | KeyCode.DownArrow],
+			mac: {
+				primary: KeyMod.Alt | KeyMod.CtrlCmd | KeyCode.RightArrow,
+				secondary: [KeyMod.Alt | KeyMod.CtrlCmd | KeyCode.DownArrow]
+			},
+			when: TerminalContextKeys.focus,
+			weight: KeybindingWeight.WorkbenchContrib
+		},
+		precondition: ContextKeyExpr.or(TerminalContextKeys.processSupported, TerminalContextKeys.terminalHasBeenCreated),
+		run: async (c) => {
+			c.groupService.activeGroup?.focusNextPane();
+			await c.groupService.showPanel(true);
+		}
+	});
+
+	registerActiveInstanceAction({
+		id: TerminalCommandId.RunRecentCommand,
+		title: { value: localize('workbench.action.terminal.runRecentCommand', "Run Recent Command..."), original: 'Run Recent Command...' },
+		precondition: ContextKeyExpr.or(TerminalContextKeys.processSupported, TerminalContextKeys.terminalHasBeenCreated),
+		keybinding: [
+			{
+				primary: KeyMod.CtrlCmd | KeyCode.KeyR,
+				mac: { primary: KeyMod.WinCtrl | KeyCode.KeyR },
+				when: ContextKeyExpr.and(TerminalContextKeys.focus, CONTEXT_ACCESSIBILITY_MODE_ENABLED),
+				weight: KeybindingWeight.WorkbenchContrib
+			},
+			{
+				primary: KeyMod.CtrlCmd | KeyMod.Alt | KeyCode.KeyR,
+				mac: { primary: KeyMod.WinCtrl | KeyMod.Alt | KeyCode.KeyR },
+				when: ContextKeyExpr.and(TerminalContextKeys.focus, CONTEXT_ACCESSIBILITY_MODE_ENABLED.negate()),
+				weight: KeybindingWeight.WorkbenchContrib
+			}
+		],
+		run: async (activeInstance, c) => {
+			await activeInstance.runRecent('command');
+			if (activeInstance?.target === TerminalLocation.Editor) {
+				await c.editorService.revealActiveEditor();
+			} else {
+				await c.groupService.showPanel(false);
 			}
 		}
 	});
-	registerAction2(class extends Action2 {
-		constructor() {
-			super({
-				id: TerminalCommandId.CopyLastCommandOutput,
-				title: { value: localize('workbench.action.terminal.copyLastCommand', 'Copy Last Command Output'), original: 'Copy Last Command Output' },
-				f1: true,
-				category,
-				precondition: ContextKeyExpr.or(TerminalContextKeys.processSupported, TerminalContextKeys.terminalHasBeenCreated)
-			});
-		}
-		async run(accessor: ServicesAccessor): Promise<void> {
-			const instance = accessor.get(ITerminalService).activeInstance;
-			const commands = instance?.capabilities.get(TerminalCapability.CommandDetection)?.commands;
+
+	registerActiveInstanceAction({
+		id: TerminalCommandId.CopyLastCommandOutput,
+		title: { value: localize('workbench.action.terminal.copyLastCommand', 'Copy Last Command Output'), original: 'Copy Last Command Output' },
+		precondition: ContextKeyExpr.or(TerminalContextKeys.processSupported, TerminalContextKeys.terminalHasBeenCreated),
+		run: async (instance, c, accessor) => {
```

Comment:
```suggestion
		run: async (instance, accessor) => {
```

Reply:
or replace all `accessor` with `c` for consistency

Reply:
c and accessor are both arguments passed in. c is preloaded with just terminal services which is what the vast majority of things use, accessor is available for any non-terminal service